service/controller/v5/cloudconfig/cloud_config.go

```diff
@@ -207,19 +246,48 @@ func (c CloudConfig) NewWorkerCloudConfig(customObject providerv1alpha1.AzureCon
 					},
 				},
 			},
-		},
-		Extension: &workerExtension{
+		}
+		params.Extension = &workerExtension{
 			Azure:         c.azure,
 			AzureConfig:   c.azureConfig,
 			CertsSearcher: c.certsSearcher,
 			CustomObject:  customObject,
-		},
-		SSOPublicKey: c.ssoPublicKey,
+			Encrypter:     c.encrypter,
+		}
+		params.SSOPublicKey = c.ssoPublicKey
+
+		ignitionPath := k8scloudconfig.GetIgnitionPath(c.ignitionPath)
+		params.Files, err = k8scloudconfig.RenderFiles(ignitionPath, params)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
 	}
 
 	return newCloudConfig(k8scloudconfig.WorkerTemplate, params)
 }
 
+// certFiles returns list of certificates paths for particular role.
+// It is used for rendering cert-decrypter systemd unit.
+func (c CloudConfig) certFiles(customObject providerv1alpha1.AzureConfig, role string) (certs.Files, error) {
+	var certFiles certs.Files
+
+	clusterCerts, err := c.certsSearcher.SearchCluster(key.ClusterID(customObject))
```

Comment:
The `cloudconfig` service should not rely on external dependencies IMO. It should be configured with data and render templates. Making the templating communicate over network smells and usually implies that network calls are done multiple times where they should only be done once. I suggest to provide the certs via method call arguments like we do in AWS and KVM.